src/extension/utils.js

```diff
@@ -666,6 +668,83 @@ export async function setProject(project, cb) {
   }
 }
 
+/**
+ * Sets the authorization header for all requests to protected sites
+ */
+export async function updateSiteAuthorizationHeaderRules() {
+  try {
+    // remove all rules first
+    await chrome.declarativeNetRequest.updateSessionRules({
+      removeRuleIds: (await chrome.declarativeNetRequest.getSessionRules())
+        .filter((rule) => rule.id >= 100)
+        .map((rule) => rule.id),
+    });
+
+    // start site authorization header rules from id 100
+    let id = 100;
+    const projects = await getConfig('local', 'hlxProtectedProjects') || [];
+    const addRules = [];
+
+    for (const [key, value] of Object.entries(projects)) {
+      const [owner, repo] = key.split('/');
+      addRules.push({
+        id,
+        priority: 1,
+        action: {
+          type: 'modifyHeaders',
+          requestHeaders: [{
+            operation: 'set',
+            header: 'authorization',
+            value: `token ${value}`,
+          }],
+        },
+        condition: {
+          regexFilter: `^https://[a-z0-9-]+--${repo}--${owner}.aem.(page|live)/.*`,
```

Comment:
Not 100% sure on `[a-z0-9-]+` for the ref.

Reply:
I think that covers it.